src/ResourceManager/Websites/Commands.Websites/Models.WebApp/WebAppOptionalSlotBaseCmdlet.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Management.Automation;
+using Microsoft.Azure.Commands.WebApps.Models;
+using Microsoft.Azure.Commands.WebApps.Utilities;
+using Microsoft.Azure.Management.WebSites.Models;
+
+namespace Microsoft.Azure.Commands.WebApps
+{
+    // For some cmdlets, the slot is optional, but will be used if specified.
+    public class WebAppOptionalSlotBaseCmdlet : WebAppBaseClientCmdLet
+    {
+        protected const string ParameterSet1Name = "S1";
+        protected const string ParameterSet2Name = "S2";
+
+        [Parameter(ParameterSetName = ParameterSet1Name, Position = 0, Mandatory = true,
+            HelpMessage = "The name of the resource group.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSet1Name, Position = 1, Mandatory = true,
+            HelpMessage = "The name of the web app.", ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSet1Name, Position = 2, Mandatory = false,
+            HelpMessage = "The name of the web app slot.")]
+        public string Slot { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSet2Name, Position = 0, Mandatory = true,
+            HelpMessage = "The web app object", ValueFromPipeline = true)]
+        [ValidateNotNullOrEmpty]
+        public Site WebApp { get; set; }
```

Comment:
There are 2 parameter sets, one using strings to name the resource, and another using a Site object. Users are expected to pipe in a Site object returned by Get-AzureRmWebApp.